src/coreclr/System.Private.CoreLib/src/System/Attribute.CoreCLR.cs

```diff
@@ -515,10 +518,11 @@ public static bool IsDefined(MemberInfo element, Type attributeType, bool inheri
             if (attrib == null || attrib.Length == 0)
                 return null;
 
+            Attribute match = attrib[0];
             if (attrib.Length == 1)
-                return attrib[0];
+                return match;
 
-            throw new AmbiguousMatchException(SR.RFLCT_AmbigCust);
+            throw GetAmbiguousMatchException(match);
```

Comment:
Here and all similar cases in this file
```suggestion
            throw ThrowHelper.GetAmbiguousMatchException(match);
```